File: codegen/results/v002/m3-max-48gb/ollama/qwen3.6_35b_think/20260418_184802/typescript/008_prime_numbers/solution.ts
function main() {
  const nStr = process.argv[2];
  if (!nStr) return;

  const n = parseInt(nStr, 10);
  if (isNaN(n) || n < 2) return;

  // Sieve of Eratosthenes
  const isPrime = new Array(n + 1).fill(true);
  isPrime[0] = false;
  isPrime[1] = false;

  for (let p = 2; p * p <= n; p++) {
    if (isPrime[p]) {
      for (let i = p * p; i <= n; i += p) {
        isPrime[i] = false;
      }
    }
  }

  const primes: number[] = [];
  for (let p = 2; p <= n; p++) {
    if (isPrime[p]) {
      primes.push(p);
    }
  }

  if (primes.length > 0) {
    console.log(primes.join('\n'));
  }
}

main();

File: codegen/results/v002/m3-max-48gb/ollama/qwen3.6_35b_think/20260418_184802/go/008_prime_numbers/solution.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	// Ensure exactly one argument is provided
	if len(os.Args) != 2 {
		return
	}

	// Parse the command line argument
	n, err := strconv.Atoi(os.Args[1])
	if err != nil || n < 2 {
		// No primes to print if N < 2 or invalid input
		return
	}

	// Sieve of Eratosthenes to find all primes up to N
	isPrime := make([]bool, n+1)
	for i := 2; i <= n; i++ {
		isPrime[i] = true
	}

	for i := 2; i*i <= n; i++ {
		if isPrime[i] {
			for j := i * i; j <= n; j += i {
				isPrime[j] = false
			}
		}
	}

	// Print each prime number on its own line
	for i := 2; i <= n; i++ {
		if isPrime[i] {
			fmt.Println(i)
		}
	}
}